packages/core/src/messages/challenge.ts

```diff
@@ -0,0 +1,61 @@
+import { u8aEquals, SECP256K1_CONSTANTS } from '@hoprnet/hopr-utils'
+import { HASH_ALGORITHM } from './constants'
+import { ecdsaSign, ecdsaVerify, publicKeyCreate } from 'secp256k1'
+import { createHash } from 'crypto'
+
+import type PeerId from 'peer-id'
+
+export class Challenge {
```

Comment:
Suggest we move this to utils.

Reply:
Postponing this because it creates chaos in utils